README.md

```diff
@@ -63,86 +62,85 @@ Must be a list of *tensor specification keys*.
+
 - `kwargs`
 Keyword arguments for the implementation specified by [`source`](#source).
+
 - `covers`
 A list of cover images provided by either a relative path to the model folder, or a hyperlink starts with `https`.\
 Please use an image smaller than 500KB, aspect ratio width to height 2:1. The supported image formats are: `jpg`, `png`, `gif`.
 <!--- `I am not quite sure what we decided on for the uri identifiers in the end, I am sticking with the simplest option for now <format>+<protocoll>://<path>`, e.g.: `conda+file://./req.txt` -->  
+
 - `dependencies` Dependency manager and dependency file, specified as `<dependency manager>:<relative path to file>`\
 For example:
   - conda:./environment.yaml
   - maven:./pom.xml
   - pip:./requirements.txt
 
-- `weights`
-A list of weights, each weights definition contains the following fields:
-    - `id` a unique id which will be used to refer to the weights. <!-- maybe with special values like 'default'? -->
-    - `name` the name of the weights for display, it should be a human-friendly name in title case
-    - `description` description about the weights, it is recommended to describe the how the weights is trained, and what's the dataset used for training.
-    - `authors` a list of authors. This field is optional, only required if the authors are different from the model.
-    - `covers` a list of cover images (see `model`:`covers`). This is used for showing how inputs and outputs look like with this weights file.
-    - `source` link to the weights file. Preferably an url to the weights file.
-    - `sha256` SHA256 checksum of the model weight file specified by `source` (see `models` section above for how to generate SHA256 checksum)
-    - `timestamp` timestamp according to [ISO 8601](#https://en.wikipedia.org/wiki/ISO_8601)
-    - `test_inputs` list of URIs to test inputs as described in `inputs` for a single test case. Supported file formats/extensions: `.npy`
-    - `test_outputs` analog to `test_inputs`.
-    - `documentation` relative path to file with additional documentation in markdown.
-    - `tags` a list of tags.
-    - `attachments` text keys and URI values to additional, relevant files.
+- `attachments` Additional files for this specification; e.g. images that are necessary for the documentation. These files will be included when generating the model package. This field is optional.
+
+- `test_inputs` list of URIs to test inputs as described in inputs for a single test case. Supported file formats/extensions: .npy
+- `test_outputs` analog to test_inputs.
+
+- `sample_inputs` list of URIs to sample inputs to illustrate possible inputs for the model, for example stored as png or tif images. This field is optional. 
+- `sample_outputs` list of URIs to sample outputs corresponding to the `sample_inputs`. This field is optional.
 
+- `weights` The weights for this model. Weights can be given for different formats, but should otherwise be equivalent.
```

Comment:
should we maybe add `kwargs` to a weights entry? E.g. for the TensorFlow SavedModelBundle one has to specify a `tag` when loading the graph (which is commonly `serve`). What's in the `kwargs` could be specified depending on the `weights_format`.

Reply:
I think you can define your own required/optional keys contained in each weights format, we just need to document them in each weight format definition.

So in that case, I would directly add a `tag` key under the weight format, then document that the `tag` key is required for your tensorflow format.

Reply:
I think the suggestion by @oeway is good to have custom keys for the weight formats and then have `tag` as a requirement for the tf weight format.

Reply:
If we agree on this, I will add some description saying that additional fields for the different formats are allowed and we can work on defining the formats in #43.